### DISCUSSION TITLE
Bind a bb ID to an integer in a SQL query

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#6311] [Help] [answered]
I try to pass my ID from the screen to the data provider using {{ URL.id }}

I get : invalid input syntax for integer : "['22']"

How can I simply put the brackets off to pass only the integer 22 ?

**Comment by NicolasFabre81** [comment]
Related with https://github.com/Budibase/budibase/issues/2417 and https://github.com/Budibase/budibase/pull/3072

Did not manage to make the workaround with handlebars...

**Comment by melohagan** [comment]
Hi @NicolasFabre81 

I wonder if a JavaScript binding might work better here. 

If it is an array, then you might be able to do `return $("URL.id")[0]`

Otherwise if it's actually a string then maybe regex `return $("URL.id").replace(/[\['\]]/g, "")`

**Comment by NicolasFabre81** [comment]
The solution with return $("URL.id").replace(/[\['\]]/g, "") works in the app !
But... not in the designer !
I get a JS error and "no rows found" in my data provider so I don't see the form elements anymore...

**Comment by NicolasFabre81** [comment, marked as answer]
Here is the final solution (22 is an existing ID) : 
`if ($("URL.id")===undefined) {
	return('22')
}
else {
	return $("URL.id").replace(/[\['\]]/g, "")
}`

**Reply by melohagan** [reply]
Nice one! You can also do: `return $("URL.id")?.replace(/[\['\]]/g, "")`

https://ponyfoo.com/articles/null-propagation-operator
